### PR TITLE
Iter optimization clean up

### PR DESCRIPTION
**Pull request by montrie**
### This PR is a cleaned up version of PR #964 

As discussed, I created a new PR containing only the relevant Iter related changes. This PR is now based on the merged `flambda` PR #928. 
In particular, it is based on the commit [https://github.com/goblint/analyzer/commit/ab2141a00b577cb20d354878a315a2e27e5074c6](url)
.
All relevant information can be found in the comments of #964.

### CODE REVIEW

**Comment by sim642** [comment]
If this subsumes #964, then feel free to close that.

Is there a plan to do a similar rebase with #965?

**Comment by montrie** [comment]
Yes, there should be a similar rebase of #965 within the next days.

**Comment by KhalilCK** [comment]
Yes,https://github.com/goblint/analyzer/pull/975 is the rebase of https://github.com/goblint/analyzer/pull/965

File: src/analyses/base.ml [hunk 1]
```diff
@@ -26,6 +26,8 @@ module Dep    = BaseDomain.PartDeps
 module WeakUpdates   = BaseDomain.WeakUpdates
 module BaseComponents = BaseDomain.BaseComponents
 
+open Iter
```
Comment (michael-schwarz) [inline src/analyses/base.ml:29]: As far as I can see, this open is unused?

File: src/analyses/base.ml [hunk 2]
```diff
@@ -1251,13 +1252,13 @@ struct
...
         match eval_rv_address (Analyses.ask_of_ctx ctx) ctx.global ctx.local e with
         | `Address a ->
-          let slen = List.map String.length (AD.to_string a) in
           let lenOf = function
             | TArray (_, l, _) -> (try Some (lenOfArray l) with LenOfArray -> None)
             | _ -> None
           in
-          let alen = List.filter_map (fun v -> lenOf v.vtype) (AD.to_var_may a) in
-          let d = List.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) (List.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int) (slen @ alen)) in
+          let d = Iter.((Iter.map String.length (Iter.of_list @@ AD.to_string a)) <+> (Iter.filter_map (fun v -> lenOf v.vtype) (Iter.of_list @@ AD.to_var_may a)) (*<- equal to slen @ alen *)
```
Comment (michael-schwarz) [inline src/analyses/base.ml:1259]: If you already have `Iter.(...)` outside, you do not need it inside again.

**Review by michael-schwarz — COMMENTED** [review]
Thanks for the PR! As discussed on the other PR, we will likely merge the alternative #975 though :)

File: src/analyses/base.ml (hunk 1 above)
Comment (montrie) [reply src/analyses/base.ml:29]: Yes, that is correct, this open isn't needed.

File: src/analyses/base.ml (hunk 2 above)
Comment (montrie) [reply src/analyses/base.ml:1259]: Right, that somehow slipped my mind. Should I push a fix for that even though the code deadline has passed?

**Comment by michael-schwarz** [comment]
As we will not move ahead with Iter, closing this.

**Pull request closed without merging**